web/src/settings_invites.ts

```diff
@@ -318,6 +386,92 @@ export function on_load_success(
+            invite_as_options: settings_config.user_role_values,
+            is_admin: current_user.is_admin,
+            is_owner: current_user.is_owner,
+            show_select_default_streams_option: stream_data.get_default_stream_ids().length > 0,
+            can_subscribe_other_users:
+                settings_data.can_subscribe_others_to_all_accessible_streams(),
+        });
+
+        function edit_invite_user_modal_post_render(): void {
+            const $submit_button = $("#edit-invite-form .dialog_submit_button");
+            $submit_button.prop("disabled", true);
+            $submit_button.attr("data-loading-text", $t({defaultMessage: "Updating link…"}));
+            if (invite?.is_multiuse) {
+                const $stream_pill_container = $("#invite_streams_container .pill-container");
+                if (invite.include_realm_default_subscriptions) {
+                    $("#invite_select_default_streams").prop("checked", true);
+                    $(".add_streams_container").hide();
+                } else {
+                    $("#invite_select_default_streams").prop("checked", false);
+                    $(".add_streams_container").show();
+                }
+                stream_pill_widget = invite_stream_picker_pill.create(
+                    $stream_pill_container,
+                    invite.stream_ids,
+                    true,
+                );
+                $("#invite_as").val(invite.invited_as);
+            }
+
+            function toggle_submit_button($elem: JQuery): void {
+                if (!invite || !invite.is_multiuse) {
+                    $elem.prop("disabled", true);
+                    return;
+                }
+                const initial_streams = [...invite.stream_ids].sort();
+                let selected_streams: number[] = [];
+                if (!$("#invite_select_default_streams").prop("checked")) {
+                    selected_streams = stream_pill.get_stream_ids(stream_pill_widget).sort();
+                }
+                const is_state_unchanged =
+                    selected_streams.length === initial_streams.length &&
+                    selected_streams.every((val, index) => val === initial_streams[index]) &&
+                    Number.parseInt(String($("#invite_as").val()), 10) === invite.invited_as &&
+                    $("#invite_select_default_streams").prop("checked") ===
+                        invite.include_realm_default_subscriptions;
+
+                $elem.prop("disabled", is_state_unchanged);
+            }
+
+            $("#edit-invite-form").on("change", "#invite_as, div.input", () => {
+                e.preventDefault();
+                toggle_submit_button($submit_button);
+            });
+            $("#invite_select_default_streams").on("change", () => {
+                e.preventDefault();
+                set_streams_to_join_list_visibility();
+                toggle_submit_button($submit_button);
+            });
```

Comment:
I have not checked, but maybe we can deduplicate some code here which is used in both invite creation and edit modals.

Reply:
Like code in `toggle_submit_button` can be defined in a single function that can be called both for creating and editing invites. And similarly `post_render` function as well. Might need some variable for groups-related code since we do not allowing editing groups list.

Reply:
Did you try deduplicating this?

Reply:
Ok, seems fine for now as "submit" button is disabled for different reasons in editing and creation modals.